lib/services/sku-pack-service.js

```diff
@@ -477,63 +497,69 @@ function skuPackServiceFactory(
-            .then(function(fileContents) {
-                return [ fileContents, self.validatePack(fileContents, fromRoot) ];
-            })
-            .spread(function(fileContents) {
-                contents = fileContents;
-                if(skuid === undefined)  {
-                    var conf = JSON.parse(contents);
-                    if(!conf.hasOwnProperty('rules') || !conf.hasOwnProperty('name')) {
-                        throw new Error('rules or name is missing');
+        .then(function(fileContents) {
+            contents = fileContents;
+            return self.validatePack(fileContents, fromRoot, {
+                validateSku: !_.isUndefined(skuid)
+            });
+        })
+        .then(function() {
+            if(skuid === undefined)  {
+                var conf = JSON.parse(contents);
+                return waterline.skus.findOne({name: conf.name})
+                .then(function(entry) {
+                    if(entry) {
+                        var err = new Errors.BaseError('duplicate name found');
+                        err.status = 409;
+                        throw err;
                     }
-                    return waterline.skus.findOne({name: conf.name})
-                    .then(function(entry) {
-                        if(entry) {
-                            var err = new Errors.BaseError('duplicate name found');
-                            err.status = 409;
-                            throw err;
-                        }
-                        return waterline.skus.create({name: conf.name, rules: conf.rules})
-                        .then(function(sku) {
-                            return sku.id;
-                        });
-                    });
-                }
-                return skuid;
-            })
-            .then(function(skuID) {
-                return fs.statAsync(self.confRoot + '/' + skuID).then(function(stat) {
-                        if(stat.isDirectory())  {
-                            return self.deletePack(skuID);
-                        }
-                        return skuID;
-                    })
-                    .catch(function() {
-                        return skuID;
-                    });
-            })
-            .then(function(skuID) {
-                dst = self.confRoot + '/' + skuID;
-                installSKU = skuID;
-                return [skuID, fs.readdirAsync(fromRoot), fs.mkdirAsync(dst) ];
-            })
-            .spread(function(sku, entries) {
-                return Promise.map(entries, function(entry) {
-                    var src = fromRoot + '/' + entry;
-                    if( entry === 'config.json' ) {
-                        return fs.renameAsync(src, dst + '.json');
-                    } else {
-                        return fs.renameAsync(src, dst + '/' + entry);
+                    return waterline.skus.create(_.pick(conf, ['name', 'rules', 'discoveryGraphName', 'discoveryGraphOptions'] ));
```

Comment:
Why only pick these keys? From your SKU model definition in on-core (https://github.com/RackHD/on-core/pull/123/files) , other keys such as "workflowRoot", "taskRoot" should also be inserted.

So either PR need a fix.